code/game/machinery/doors/airlock.dm

```diff
@@ -1731,6 +1742,718 @@
 	var/area/source_area = get_area(src)
 	return source_area?.airlock_wires ? new source_area.airlock_wires(src) : new /datum/wires/airlock(src)
 
+/obj/structure/fluff/airlock_filler/Initialize(mapload)
+	. = ..()
+	RegisterSignal(filled_airlock, COMSIG_QDELETING, PROC_REF(no_airlock))
```

Comment:
So this I think won't work because `filled_airlock` is null on initialize.
I think you either want to pass `filled_airlock` into the constructor when you are creating this object, or make a new proc which sets `filled_airlock` to the passed airlock reference and _then_ registers the signal, which is called immediately after you create it (where you currently set its value).

Reply:
Okay, I added it to a proc when the airlock creates a filler.